src/main/js/pullRequest.ts

```diff
@@ -22,13 +22,170 @@
  * SOFTWARE.
  */
 
-import { BasicComment, BasicPullRequest, MergeCommit, PossibleTransition, PullRequest } from "./types/PullRequest";
-import { apiClient } from "@scm-manager/ui-components";
-import { Branch, Link, Repository } from "@scm-manager/ui-types";
+import {
+  BasicComment,
+  BasicPullRequest,
+  CheckResult,
+  Comment,
+  Comments,
+  Conflicts,
+  MergeCheck,
+  MergeCommit,
+  PossibleTransition,
+  PullRequest,
+  PullRequestCollection
+} from "./types/PullRequest";
+import { apiClient, ConflictError, NotFoundError } from "@scm-manager/ui-components";
+import { Changeset, HalRepresentation, Link, PagedCollection, Repository } from "@scm-manager/ui-types";
+import { useMutation, useQuery, useQueryClient } from "react-query";
+import { MergeStrategyInfo } from "./types/MergeStrategyInfo";
 
 const CONTENT_TYPE_PULLREQUEST = "application/vnd.scmm-pullRequest+json;v=2";
 
-export function createPullRequest(url: string, pullRequest: BasicPullRequest) {
+// React-Query Hooks
+
+export const invalidatePullRequest = (repository: Repository, pullRequestId?: string) => {
+  if (!pullRequestId) {
+    throw new Error("could not invalidate pull request without id");
+  }
+
+  const queryClient = useQueryClient();
+  queryClient.invalidateQueries(prQueryKey(repository, pullRequestId));
+};
+
+export const prQueryKey = (repository: Repository, pullRequestId: string) => {
+  return ["repository", repository.namespace, repository.name, "pull-request", pullRequestId];
+};
+
+const prsQueryKey = (repository: Repository, status?: string) => {
+  return ["repository", repository.namespace, repository.name, "pull-requests", status || ""];
+};
+
+const prCommentsQueryKey = (repository: Repository, pullRequest: PullRequest) => {
+  return ["repository", repository.namespace, repository.name, "pull-request", pullRequest.id, "comments"];
+};
+
+export const usePullRequest = (repository: Repository, pullRequestId?: string) => {
+  if (!pullRequestId) {
+    throw new Error("Could not fetch pull request without id");
+  }
+
+  if (!repository._links.pullRequest) {
+    throw new Error("Could not fetch pull request because link is missing");
+  }
+
+  const { error, isLoading, data } = useQuery<PullRequest, Error>(prQueryKey(repository, pullRequestId), () =>
+    apiClient.get((repository._links.pullRequest as Link).href + "/" + pullRequestId).then(response => response.json())
+  );
+
+  return {
+    error,
+    isLoading,
+    data
+  };
+};
+
+export const useUpdatePullRequest = (repository: Repository, pullRequest: PullRequest, callback?: () => void) => {
+  const id = pullRequest.id;
+
+  if (!id) {
+    throw new Error("Could not update pull request without id");
+  }
+
+  const queryClient = useQueryClient();
+  const { isLoading, error, mutate } = useMutation<unknown, Error, PullRequest>(
+    pr => {
+      if (!pr._links.update) {
+        throw new Error("Could not update pull request because update link is missing");
+      }
```

Comment:
Looks like a pattern ... `requiredLink(halObject, link)` throws error or returns the href?